### PR TITLE
fix: properly check friend list limits

### LINKED ISSUE #798: BUG: Clients can bypass friend limits by sending requeta from someone below the limit
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

Main

### Issue Description

Title

### Reproduction Steps

Have a full friends list and a second player without a full list.  
Player 2 sends a request to player one and they can accept.

### Expected Behavior

Request sends the property response code since a friend's list is full.

### Environment

All

### PR DESCRIPTION
**Pull request by aronwk-aaron**
added a config for friend list limit for the brave that want to mod the client to sanely go over 50 
moved the best friend limit config to chatconfig.ini where it should be 
cleanup loading these configs options a bit

Tested that the BFF limit works still and that the new friend limit works as well
fixes #798

### CODE REVIEW

File: dChatServer/PlayerContainer.h [hunk 1]
```diff
@@ -75,5 +77,7 @@ class PlayerContainer {
 	std::map<LWOOBJID, PlayerData*> mPlayers;
 	std::vector<TeamData*> mTeams;
 	std::unordered_map<LWOOBJID, std::u16string> mNames;
+	uint32_t mMaxNumberOfBestFriends = 5;
```
Comment (EmosewaMC) [inline dChatServer/PlayerContainer.h:80]: Let's at least be consistent now with m_

Comment (aronwk-aaron) [reply dChatServer/PlayerContainer.h:80]: I followed the format of this file, want me to change the rest of them as well?

Comment (EmosewaMC) [reply dChatServer/PlayerContainer.h]: Nah just going forward would be good

**Review by EmosewaMC — APPROVED** [review]
Oh you did all of them lol

**Review by Jettford — APPROVED** [review]

**Pull request merged**